language/diem-framework/modules/CRSN.move

```diff
@@ -24,8 +26,23 @@ module DiemFramework::CRSN {
         min_nonce: u64,
         size: u64,
         slots: BitVector,
+        force_shift_events: EventHandle<ForceShiftEvent>,
     }
 
+    /// Whenever a force shift is performed a `ForceShiftEvent` is emitted
+    struct ForceShiftEvent has drop, store {
+        /// current LHS
+        current_min_nonce: u64,
+        /// The amount the window is being shifted
+        shift_amount: u64,
+        /// The state of the bitvector just before the shift (NB: the sequence
+        /// nonce of the shifting transaction will not be set).
+        bits_at_shift: BitVector,
```

Comment:
random thought: is there any reason why the BitVector uses a `vector<bool>`? do we have a specialized compact serialization for that?

Reply:
nit: also maybe add a small comment on why we need the `bits_at_shift` here, like "the bitvector state is needed to prove that a CRSN slot was expired and not already consumed; therefore, a user transaction at that slot can't exist (because that slot was expired)."

Reply:
> random thought: is there any reason why the BitVector uses a vector<bool>? do we have a specialized compact serialization for that?

Good question. The underlying bitvector representation can be compacted (e.g., to use vec<u64>), however verifying code that does computation around bitshifts is incredibly difficult, so there are issues with a more complex dense representation of these and we decided to go with vec<bool> for now.